staging/src/k8s.io/apimachinery/pkg/runtime/error.go

```diff
@@ -120,3 +120,77 @@ func IsMissingVersion(err error) bool {
 	_, ok := err.(*missingVersionErr)
 	return ok
 }
+
+// StrictDecoderError is a base error type that is returned by a strict Decoder such
+// as UniversalStrictDecoder.
+type StrictDecoderError struct {
+	message      string
+	gvk          schema.GroupVersionKind
+	originalData []byte
+}
+
+// UnknownFieldError is an error type that is returned in cases where the input
+// JSON or YAML contains unknown fields.
+// UnknownFieldError embeds StrictDecoderError.
+type UnknownFieldError struct {
+	StrictDecoderError
+}
+
+// DuplicateFieldError is an error type that is returned in cases where the input
+// JSON or YAML contains duplicate fields under the same parent field.
+// DuplicateFieldError embeds StrictDecoderError.
+type DuplicateFieldError struct {
```

Comment:
same here, I'd rather start small and expand structured errors once we actually have data to make them useful